RELEASE.md

```diff
@@ -0,0 +1,253 @@
+- The tracking issue is created using the [release-planning.md](.github/ISSUE_TEMPLATE/release-planning.md) template
+- The tracking issue is assigned to the release coordinator
+- The tracking issue is the source of truth for the current state of the release
+
+### Step 2: Development
+
+- The development phase is the first 2.5 months of the release cycle
+- During this phase the `master` branch is open for new changes
+- There are no restrictions on the type of changes that can be made to the codebase
+  during this phase
+- The normal review process is followed for all changes
+- The release coordinator updates the community on the progress of the release during the [community meeting]
+
+### Step 3: Code Freeze
+
+- The code freeze date is the 2 weeks before the release date
+- During this phase the `master` branch is closed for new changes
+- Any bug fixes or documentation changes must follow the normal review process, but will also require the explicit approval of the release coordinator.
+- OKEPs may be reviewed and merged into the `master` branch during this phase without explicit approval from the release coordinator since they do not affect the release itself.
+
+### Step 4: (Optional) Release Candidate
+
+At the discretion of the release coordinator and the maintainers, a release candidate may be created.
+The decision to create a release candidate is made at the [community meeting] after the code freeze date.
+
+- Once the master branch is closed, the release coordinator may tag a new pre-release
+- The pre-release is named `v<release-number>-rc<rc-number>`
+- The pre-release is created from the `master` branch
+
+Pre-release versions may be subjected to additional testing and review by the community.
+
+### Step 5: Release
+
+- Once the release date is reached and all required changes have been merged into the `master` branch, the release coordinator will tag a new release with the version `v<release-number>`.
+- The release is created from the `master` branch
+- The release coordinator creates a GitHub release with release notes
+- Release notes should include:
+  - Summary of new features
+  - List of bug fixes
+  - Breaking changes (if any)
+  - Known issues
+  - Upgrade notes
+
+### Step 6: Post-Release
+
+- The release coordinator announces the release to the community via:
+  - GitHub release notes
+  - Community Slack channels
+  - Mailing list
+- The release coordinator updates the tracking issue to mark the release as complete
+- The release coordinator creates a retrospective document to capture lessons learned
+
+## Release Communication
+
+Throughout the release process, the release coordinator should maintain clear communication with:
+
+- **Maintainers**: For technical decisions and code review coordination
+- **Community**: For progress updates and feedback solicitation
+- **Stakeholders**: For timeline and feature delivery expectations
```

Comment:
does it make sense to "define" these roles in the beginning of this doc ? 
I don't have a strong opinion here.